File: Selection_sort.cpp
#include <iostream>
using namespace std;
int main() {

int array[]={9,7,3,1,6};
cout<<"before sorting the values : ";
for ( int i =0;i<5;i++){
    cout<<array[i]<<" ";
}
cout<<endl;
int n = sizeof(array)/sizeof(array[0]);
cout<<"value after sorting the array : ";

for( int i = 0;i<n-1;i++){
int index = i ;  
for( int j =i+1; j<n;j++){
    
    if(array[j]<array[index])
    index = j; 
}

swap(array[ index],array[i]);
}

for(int i =0;i<n;i++){
    cout<<array[i]<<" ";
}
    return 0;
}

File: Selection_sort.java
public class HelloWorld {
    
    public static void swap(int[] arr, int x, int y) {
        int temp = arr[x];
        arr[x] = arr[y];
        arr[y] = temp;
    }

    public static void main(String[] args) {
        int[] arr = {5, 6, 3, 4, 2};
        int n = arr.length;

        System.out.print("Array before sorting: ");
        for(int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();

        for(int i = 0; i < n - 1; i++) {
            int index = i;
            for(int j = i + 1; j < n; j++) {
                if(arr[j] < arr[index]) {
                    index = j;
                }
            }
            swap(arr, index, i);
        }

        System.out.print("Array after sorting: ");
        for(int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}
